src/tool/ToolArgs.cpp

```diff
@@ -84,6 +84,20 @@ R"(
     "inputs",
     llvm::cl::Sink,
     llvm::cl::desc("The path to the compilation database."))
+
+, generateCompilationDatabase(
```

Comment:
Do we need that option? If the input path is a project directory instead of a compile_commands.json path, we can just assume we need to generate the compilation database. The cases are mutually exclusive:

1) `<dir>/compile_commands.json`: old behavior
2) `<dir>/` (a directory): run cmake with dir as source and (1)
3) `<dir>/CMakeLists.txt`: get parent <dir> and (2)
4) no input (?): get current working directory as <dir> and (2)

Reply:
I've implemented the requested refactoring for automatic compilation database generation, covering scenarios 1-3 as discussed.

Regarding scenario 4 (no input provided): I've held back on using the current working directory for output generation. I believe we should avoid cluttering the project directory with temporary files. Your input on this would be valuable.

Most of Alan's comments have been addressed, with a few remaining which I'm working on.